hbase-server/src/main/java/org/apache/hadoop/hbase/namequeues/LogEventHandler.java

```diff
@@ -50,64 +50,83 @@
 import org.apache.hadoop.hbase.shaded.protobuf.generated.TooSlowLog.SlowLogPayload;
 
 /**
- * Event Handler run by disruptor ringbuffer consumer
+ * Event Handler run by disruptor ringbuffer consumer.
+ * Although this is generic implementation for namedQueue, it can have individual queue specific
+ * logic.
  */
 @InterfaceAudience.Private
 class LogEventHandler implements EventHandler<RingBufferEnvelope> {
 
   private static final Logger LOG = LoggerFactory.getLogger(LogEventHandler.class);
 
-  private static final String SYS_TABLE_QUEUE_SIZE =
-    "hbase.regionserver.slowlog.systable.queue.size";
-  private static final int DEFAULT_SYS_TABLE_QUEUE_SIZE = 1000;
-  private static final int SYSTABLE_PUT_BATCH_SIZE = 100;
+  private static final String SLOW_LOG_RING_BUFFER_SIZE =
+    "hbase.regionserver.slowlog.ringbuffer.size";
+
+  // Map that binds namedQueues.
+  // If NamedQueue of specific type is enabled, corresponding Queue will be used to
+  // insert and retrieve records.
+  // Individual queue sizes should be determined based on their individual configs.
+  private final Map<NamedQueuePayload.NamedQueueEvent, Queue> namedQueues = new HashMap<>();
```

Comment:
For example, the caller should be able to determine which queue implementations (and its related services) should be loaded. That would isolate LogEventHandler from being constantly modified by additional services requiring this type of loggings.

Reply:
Strategy pattern is good thought but we don't have common payload coming from client so far. Specifically when admin client will talk to RSRpcServer and that will talk to `LogEventHandler`, they expect specific payload for specific use-case. However, this fits well for strategy pattern and we should have common request payload. Let me try this one. Thanks for the suggestion @wchevreuil .